assets/js/datepicker.js

```diff
@@ -1,27 +1,37 @@
 /* global job_manager_datepicker */
-jQuery(document).ready(function($) {
+jQuery(document).on( 'ready', function() {
```

Comment:
So regarding $ vs JQuery, I don't think it makes much difference. I have a slight preferance for JQuery as it avoids potential conflict (although conflicts are not possible in this case). So the change was done because I wanted to subscribe to many events but this isn't the case anymore.

The problem however was the usage of `on('ready')` which although it runs, it causes a weird initialization issue. Although `wp_localize_jquery_ui_datepicker` was running before `datepicker.js`, it seemed like the datepicker wasn't initialized during the initial page load. This caused the issue that you mentioned in your other comment which should be fixed now. 59fc67dfa704f737e6488c277cd94b15b0d05675

Reply:
> So regarding $ vs JQuery, I don't think it makes much difference. I have a slight preferance for JQuery as it avoids potential conflicts (although conflicts were not possible in this case). So the change was done because I wanted to subscribe to many events in a previous commit.

I see. For me it is pretty indifferent, too. However, I do like to maintain a pattern in the code, as it is easier to me to look at a code and know what to expect according to the rest of the project, so maintaining the usage of `jQuery` vs. `$` in a consistent way helps me a lot.

> The problem however was the usage of on('ready') which although it runs, it causes a weird initialization issue. Although wp_localize_jquery_ui_datepicker was running before datepicker.js, it seemed like the datepicker wasn't initialized during the initial page load. This caused the issue that you mentioned in your other comment which should be fixed now. https://github.com/Automattic/WP-Job-Manager/commit/59fc67dfa704f737e6488c277cd94b15b0d05675

I can confirm that this fixes the issue. Thanks!